include/xtensor/xarray.hpp

```diff
@@ -515,7 +515,7 @@ namespace xt
         base_type::shape_impl() = std::move(const_cast<shape_type&>(rhs.shape()));
         base_type::strides_impl() = std::move(const_cast<strides_type&>(rhs.strides()));
         base_type::backstrides_impl() = std::move(const_cast<backstrides_type&>(rhs.backstrides()));
-        m_data = std::move(rhs.data());
+        m_data = xtl::forward_sequence<container_type>(std::move(rhs.data()));
```

Comment:
Using `std::move` alone is ok. See the definition of `xcontainer_inner_types`:
```cpp
template <class EC, layout_type L, class SC, class Tag>
    struct xcontainer_inner_types<xarray_adaptor<EC, L, SC, Tag>>
    {
        using container_type = std::remove_reference_t<EC>;
        // ....
        using temporary_type = xarray_container<temporary_container_t<container_type>, L, SC, Tag>;
        static constexpr layout_type layout = L;
    };
```

`temporary_type_t<container_type>` differs from `container_type` only for `xbuffer_adaptor`, and that latter defines a move constructor taking a `temporary_container` parameter.

So this changed should be reverted.